admin/src/Login.js

```diff
@@ -30,21 +55,26 @@ export const Login = ({ token, secret, onSecretChange, onTokenChange, submit })
 );
 
 Login.propTypes = {
+    projectId: PropTypes.string,
     token: PropTypes.string,
     secret: PropTypes.string,
+    error: PropTypes.string,
+    onProjectIdChange: PropTypes.func.isRequired,
     onSecretChange: PropTypes.func.isRequired,
     onTokenChange: PropTypes.func.isRequired,
     submit: PropTypes.func.isRequired,
 };
 
 export default provideLoginState(
-    injectState(({ state, effects }) => (
+    injectState(({ state: { projectId, token, secret }, effects }) => (
         <Login
-            token={state.token}
-            secret={state.secret}
+            projectId={projectId}
+            token={token}
+            secret={secret}
             onSecretChange={effects.onSecretChange}
             onTokenChange={effects.onTokenChange}
-            submit={effects.submit}
+            onProjectIdChange={effects.onProjectIdChange}
+            submit={() => effects.submit({ projectId, token, secret })}
```

Comment:
This is bad, you declare a function inside jsx. Take a look at my pr to see how you can initialize the initial state from the component props and use the state in the effects